pkg/db/dto/git_provider.go

```diff
@@ -8,28 +8,34 @@ import (
 )
 
 type GitProviderConfigDTO struct {
-	Id         string  `gorm:"primaryKey"`
-	Username   string  `json:"username"`
-	Token      string  `json:"token"`
-	BaseApiUrl *string `json:"baseApiUrl,omitempty"`
+	Id            string  `gorm:"primaryKey"`
+	Username      string  `json:"username"`
+	Token         string  `json:"token"`
+	BaseApiUrl    *string `json:"baseApiUrl,omitempty"`
+	SigningKey    string  `json:"siginingKey"`
+	SigningMethod string  `json:"siginingMethod"`
```

Comment:
This should be an enum of `ssh` or `gpg` and also be a pointer since it's optional